internal/state/module.go

```diff
@@ -988,7 +988,7 @@ func (s *ModuleStore) SetModuleDiagnosticsState(path string, source ast.Diagnost
 func (s *ModuleStore) UpdateVarsDiagnostics(path string, source ast.DiagnosticSource, diags ast.VarsDiags) error {
 	txn := s.db.Txn(true)
 	txn.Defer(func() {
-		s.SetVarsDiagnosticsState(path, ast.HCLParsingSource, op.OpStateLoaded)
+		s.SetVarsDiagnosticsState(path, source, op.OpStateLoaded)
```

Comment:
The correct source is already being passed in https://github.com/hashicorp/terraform-ls/blob/69cc1b857dc07f4c23df5f24bcf597202125f2a2/internal/terraform/module/module_ops.go#L422 so this seems to be just a bug which was never surfaced as we never had any other sources up to this point.

Reply:
Good catch!